service/src/main/java/io/camunda/service/JobServices.java

```diff
@@ -52,18 +52,24 @@ public JobServices<T> withAuthentication(final Authentication authentication) {
   }
 
   public void activateJobs(
-      final ActivateJobsRequest request,
+      final String type,
+      final int maxJobsToActivate,
+      final List<String> tenantIds,
+      final long timeout,
+      final String worker,
+      final List<String> variables,
+      final long requestTimeout,
       final ResponseObserver<T> responseObserver,
       final Consumer<Runnable> cancelationHandlerConsumer) {
```

Comment:
❓ Any specific reason to move to this long list of parameters instead of a parameter object? This is usually not recommended for an API since a long list of method parameters becomes hard to read and understand really quick.

Reply:
Yep, in the service module we don't have access to records from `RequestMapper`. I thought about it a bit, and I prefer having all the records in the same place, even if this means to have a methods with (maybe) too much args

Reply:
I'm not sure if I agree here, to be honest. I think this makes the service methods worse, which is not favorable. Let's rather have the services ALWAYS define the records they consume/produce then as a rule.

Reply:
The services know best what they need/create, so they should handle this --> separation of concerns

Reply:
If a service method doesn't justify a parameter object (e.g. just max. 3 parameters), it doesn't expose one.

Reply:
I mean, having the records in the same place doesn't mean to have them in the Mapper 🤣 
I can move them into the services, so they can be also separated by type (Job and User Task)

Reply:
Well, I think (but I might be wrong) the records in the mapper are only used in the REST controllers, they are not necessarily sent to the services as well. So, the services declaring them but not using them makes not much sense.

We have helper records in the mapper to create request data we can easily pass around in the REST classes until we forward some of their data to the services.

We also have helper records in some services if the method signatures would otherwise become too long. The REST mapper reuses those if they are already there.

I personally think this is actually OK, but that might just be me 😄

Reply:
I mean, with my changes also `ActivateJobsRequest ` record is used only in the controller 😄 I don't get your point here

Reply:
The way to fix that is by introducing a parameter object that the service class can use. Which brings us back to the code without the adjustments here. 🙂

Reply:
I'll roll back the changes 🛩️ 

But I'll keep an eye on this, because I'd like to have all the records in the same place 👍

Reply:
Thanks 👍
Please don't be disappointed if that is not possible or leads to an over-engineered solution 😅 I understand the drive behind it but the REST mapper simply is an intersection place for different "worlds" (REST data and services data). It cannot contain all possible data structures it deals with, mapping back and forth. Some of them it handles on its own (the records defined in it only needed for REST), some of them come from the services API (either records created there or even those it simply passes along from the broker results).